internal/plugins/helm/v1/scaffolds/api.go

```diff
@@ -70,26 +72,42 @@ func (s *apiScaffolder) scaffold() error {
 	if err != nil {
 		return err
 	}
-	r, chrt, err := chartutil.CreateChart(projectDir, s.opts)
+	resourceOptions, chrt, err := chartutil.CreateChart(projectDir, s.opts)
 	if err != nil {
 		return err
 	}
 
 	// Check that resource doesn't exist
-	if s.config.HasResource(r.GVK()) {
+	if s.config.HasResource(resourceOptions.GVK()) {
 		return errors.New("the API resource already exists")
 	}
+
 	// Check that the provided group can be added to the project
-	if !s.config.MultiGroup && len(s.config.Resources) != 0 && !s.config.HasGroup(r.Group) {
+	if !s.config.IsMultiGroup() && s.config.ResourcesLength() != 0 && !s.config.HasGroup(resourceOptions.Group) {
 		return errors.New("multiple groups are not allowed by default, to enable multi-group set 'multigroup: true' in your PROJECT file")
 	}
 
-	res := r.NewResource(s.config, true)
-	s.config.UpdateResources(res.GVK())
+	resource := resourceOptions.NewResource(s.config)
+	s.config.UpdateResource(resource)
+
+	domain := s.config.GetDomain()
+	resource.Domain = s.opts.GVK.Group
+	if domain != "" && s.opts.GVK.Group != "" {
+		resource.Domain += "." + domain
```

Comment:
Same as above, Domain should not contain Group, use `Resource.QualifiedGroup` if you want the full group/domain which handles empty group/domains